### ISSUE TITLE
[Bug] [opercmd] When using opercmd and the users profile does not have access READ access to MVS.MCSOPER.ZOAU , checks are overlooked

### ISSUE THREAD

**Issue by ddimatos** [ansible-collections/ibm_zos_core#1254]
### Is there an existing issue for this?

- [X] There are no existing issues.

### Are the dependencies a supported version?

- [X] The dependencies are supported.

### IBM Z Open Automation Utilities

v1.2.5

### IBM Enterprise Python

v3.12.x

### IBM z/OS Ansible core Version

v1.8.0-beta.1

### ansible-version

latest

### z/OS version

None

### Bug description

Opercmd requires that a profile be permitted to class opercmds, the docs are [here](https://www.ibm.com/docs/en/zoau/1.2.x?topic=installing-configuring-zoau). The problem is in zoau 1.2.5 if a user did not have proper authority the error was put in the zoauresponse stderr object instead of throwing an exception, all code should be updated to check the stderr in this use case. In 1.3 the code will throw an exception so its best to prepare for both. 

How to test and familiarize yourself with this, the easies path is to use an existing user on the zVM and reduce their authority.

Deny access:
```
_USER="usrt001"
SAF_PROFILE="MVS.MCSOPER.ZOAU"
SAF_CLASS="OPERCMDS"
tsocmd "RDEFINE ${SAF_CLASS} ${SAF_PROFILE} UACC(NONE) AUDIT(ALL)"
tsocmd "PERMIT ${SAF_PROFILE} CLASS(${SAF_CLASS}) ID(${_USER}) ACCESS(NONE)"
tsocmd "SETROPTS RACLIST(${SAF_CLASS}) REFRESH"
```

Try the python snippet below, review the output, notice the `stderr_response` error `BGYSC0819E`, the code needs to weary of such a use case. Be sure to either log in as the user or `su` once dialed into the zvm.
```
[ USRT001:/u/omvsadm: ]python3
Python 3.10.9 (heads/pyz_dev-3.10:31fe8baa98, Mar 21 2023, 11:36:49) [Clang 4.0.1 (tags/RELEASE_401/final)] on zos
Type "help", "copyright", "credits" or "license" for more information.
>>> from zoautil_py import opercmd
>>> opercmd.execute("d t")
<zoautil_py.types.ZOAUResponse object at 0x5008EDF310>
>>> print(resp.rc)
6
>>> pprint.pprint(vars(resp))
{'command': 'opercmd -T 0 -- "d t "',
 'rc': 6,
 'response_format': 'UTF-8',
 'stderr_response': 'BGYSC0819E Insufficient security authorization for '
                    'resource MVS.MCSOPER.ZOAU in class OPERCMDS.\n'
                    'BGYSC0820E SAF rc=0x8, RACF rc=0x8, rsn=0x0\n',
 'stdout_response': ''}
```

Using SSH:
```
[ OMVSADM:/u/omvsadm: ]opercmd "d t"
BGYSC0819E Insufficient security authorization for resource MVS.MCSOPER.ZOAU in class OPERCMDS.
BGYSC0820E SAF rc=0x8, RACF rc=0x8, rsn=0x0
```

Then restore the user's access:
```
_USER="usrt001"
SAF_PROFILE="MVS.MCSOPER.ZOAU"
SAF_CLASS="OPERCMDS"
tsocmd "PERMIT ${SAF_PROFILE} CLASS(${SAF_CLASS})  ID(${USER}) ACCESS(READ)" >/dev/null 2>&1
tsocmd "SETROPTS RACLIST(${SAF_CLASS}) REFRESH" >/dev/null 2>&1
```

See [code snippet here](https://github.com/ansible-collections/ibm_zos_core/blob/80bbc263923f92a5c719afc96cdb560995466e85/plugins/modules/zos_copy.py#L2619) as an example where the `ZOAUResponse` was not evaluated

When you run this task with the user who has no permission to run opercmd, the zos_copy code should have inspected the ZOAURepsonse object and handled the access error. Instead what happens is the code becomes a noop. 

```
    - name: Create a PDS data set if it does not exist
      zos_data_set:
        name: ANSIBLE.TEST.PDS
        type: pds
        space_primary: 5
        space_type: M
        record_format: fba
        record_length: 80
        replace: yes
      register: result
      tags: setup

    - name: Results
      debug:
        var: result
      tags: setup

    - name: Create a MEMBER
      zos_data_set:
        name: ANSIBLE.TEST.PDS(TEST11)
        state: present
        replace: true
        type: MEMBER
      register: result
      tags: version1

    - name: Results
      debug:
        var: result
      tags: version1

    - name: Copy job from controller to PSDE member
      ibm.ibm_zos_core.zos_copy:
        src: "{{playbook_dir}}/files/TYPRUNSCAN.jcl"
        dest: ANSIBLE.TEST.PDS(TEST11)
        encoding:
          from: UTF-8
          to: IBM-1047
        force: true
      register: result
      tags: version1

    - name: Results
      debug:
        var: result
      tags: version1
```

**Comment by ddimatos** [comment]
Note, in ZOAU 1.3.1 this use case where a user does not have authority to run opercmd will throw an exception, in 1.3.0 it does NOT through an exception so we must cognizant of both and code appropriately.

**Comment by ddimatos** [comment]
The other option , instead of investing in this code and needing to manage 1.3.0/1.3.1, the team could consider removing the code and investing in this other solution which would not require opercmd. #1252

**Comment by ddimatos** [comment]
This also relates to issue #1253

**Comment by ddimatos** [comment]
Likely more than one oper.execute() we need to wrap with a catch.

**Comment by ddimatos** [comment]
I can see two options to testing this;
1) A manual verification after updating the block of code to check the RC and/or zoauresponse where this is done offline by running a playbook.
2) Update the pytest zInventory fixture to allow for dynamic changing of the user, we currently have 2 fixtures supported. This would be done in the test case, it still needs exploration and was not initially in scope for the current sizing.

**Comment by ddimatos** [comment]
This is related to issue #1757 , the but is addressed in #1254 and the framework delivered in #1757 because this work was split across 2 different quarters so we created 2 issues.
